python/cuml/test/test_base.py

```diff
@@ -56,3 +67,84 @@ def test_base_n_features_in(datatype, use_integer_n_features):
     else:
         clf._set_n_features_in(X_train)
         assert clf.n_features_in_ == X_train.shape[1]
+
+
+@pytest.mark.parametrize('child_class', list(all_base_children.keys()))
+def test_base_children_init(child_class: str):
+
+    # Regex for find and replace:
+    # output_type: `^[ ]{4}output_type :.*\n(^(?![ ]{0,4}(?![ ]{4,})).*(\n))+`
+    # verbose: `^[ ]{4}verbose :.*\n(^(?![ ]{0,4}(?![ ]{4,})).*(\n))+`
+    # handle: `^[ ]{4}handle :.*\n(^(?![ ]{0,4}(?![ ]{4,})).*(\n))+`
+
+    def get_param_doc(param_doc_obj, name: str):
+        found_doc = next((x for x in param_doc_obj if x.name == name), None)
+
+        assert found_doc is not None, \
+            "Could not find {} in docstring".format(name)
+
+        return found_doc
+
+    base_sig = inspect.signature(cuml.Base, follow_wrapped=True)
```

Comment:
ok, so this is comparing docstring to the init signature? Sounds like a nice approach... since this is likely to trip when people are adding new classes or making errors in docstrings, it would be nice to add a comment explaining the test. And maybe make the name as descriptive as possible like `test_base_subclass_init_matches_docs`. There is a lot of metaprogramming necessary here, so it's nice to guide other readers a bit

Reply:
Its actually comparing the docstring parameters of the current class to the docstring parameters of the `Base` class to ensure we dont have 100 different descriptions for `handle` (like we did before). Will update the test and add more info.

Reply:
Added more description to the test. Resolved